src/components/questionary/questionaryComponents/SampleDeclaration/QuestionaryComponentSampleDeclaration.tsx

```diff
@@ -194,14 +198,26 @@ function QuestionaryComponentSampleDeclaration(props: BasicComponentProps) {
             }}
             sampleCreated={newSample => {
               const newStateValue = [...stateValue, newSample.id];
-
+              setSelectedSample(newSample);
               setStateValue(newStateValue);
               onComplete(proposalQuestionId, newStateValue);
 
               const newRows = [...rows, sampleToListRow(newSample)];
               setRows(newRows);
             }}
-            sampleEditDone={() => setSelectedSample(null)}
+            sampleEditDone={() => {
+              setSelectedSample(null);
```

Comment:
I would move this line to the end of this handler. Even though it works as is because the useState is async, I think it's more readable if done after accessing the selectedSample